docs/DafnyRef/Attributes.md

```diff
@@ -141,32 +95,131 @@ be added.
    if (F != null) { Repr := Repr + {F} + F.Repr; }
 ```
 
-### 22.1.4. axiom
+### 22.1.2. `{:nativeType}` {#sec-nativetype}
+The `{:nativeType}` attribute may only be used on a ``NewtypeDecl``
+where the base type is an integral type or a real type. For example:
+
+```dafny
+newtype {:nativeType "byte"} ubyte = x : int | 0 <= x < 256
+newtype {:nativeType "byte"} ubyte = x : int | 0 <= x < 257 // Fails
+```
+
+It can take one of the following forms:
+
+* `{:nativeType}` - With no parameters it has no effect and the ``NewtypeDecl``
+will have its default behavior which is to choose a native type that can hold any
+value satisfying the constraints, if possible, otherwise BigInteger is used.
+* `{:nativeType true}` - Also gives default ``NewtypeDecl`` behavior,
+but gives an error if base type is not integral.
+* `{:nativeType false}` - Inhibits using a native type. BigInteger is used.
+* `{:nativeType "typename"}` - This form has an native integral
+type name as a string literal. Acceptable values are:
+   * `"byte"`      8 bits, unsigned
+   * `"sbyte"`     8 bits, signed
+   * `"ushort"`    16 bits, unsigned
+   * `"short"`     16 bits, signed
+   * `"uint"`      32 bits, unsigned
+   * `"int"`       32 bits, signed
+   * `"number"`    53 bits, signed
+   * `"ulong"`     64 bits, unsigned
+   * `"long"`      64 bits, signed
+  If the target compiler
+  does not support X, then an error is generated. Also, if, after
+  scrutinizing the constraint predicate, the compiler cannot confirm
+  that the type's values will fit in X, an error is generated.
+
+### 22.1.3. `{:ignore}` (deprecated)
+Ignore the declaration (after checking for duplicate names).
+
+### 22.1.4. `{:extern}` {#sec-extern}
+
+`{:extern}`  
+`{:extern <s1:string>}`  
+`{:extern <s1:string>, <s2:string>}`  
+
+`{:extern}` is a target-language dependent modifier used:
+
+* to alter the CompileName of entities such as modules, classes, methods, etc.,
+* to alter the ReferenceName of the entities,
+* to decide how to define external opaque types,
+* to decide whether to emit target code or not, and
+* to decide whether a declaration is allowed not to have a body.
+
+The `CompileName` is the name for the entity when translating to one of the target languages.
+The `ReferenceName` is the name used to refer to the entity in the target language.
+A common use case of `{:extern}` is to avoid name clashes with existing library functions.
+
+`{:extern}` takes 0, 1, or 2 (possibly empty) string arguments:
+
+- 0 arguments: Dafny will use the Dafny name as the `CompileName` and not affect the `ReferenceName`
+- 1 arguments: Dafny will use `s1` as the `CompileName`, and replaces the last portion of the `ReferenceName` by `s1`.
```

Comment:
Your `s1` and `s2` symbols don't resolve to anything. :) I would replace "1 arguments" with "`{:extern s1}`" and likewise in the next bullet.

Reply:
Line 137 and 138 are defining s1 :-) but I agree your approach is better, let me do the changes